packages/table/src/Table.tsx

```diff
@@ -0,0 +1,196 @@
+import React, { useReducer, ReactElement } from 'react';
+import { isComponentType } from '@leafygreen-ui/lib';
+import { css } from '@leafygreen-ui/emotion';
+import HeaderRow, { HeaderRowProps } from './HeaderRow';
+import TableHeader, { TableHeaderProps } from './TableHeader';
+import CheckboxCell from './CheckboxCell';
+import { State, Types, TableProvider, reducer } from './table-context';
+
+// * Clean TS
+// * Add nodeRef to Row
+
+const tableStyles = css`
+  border-collapse: collapse;
+  box-sizing: border-box;
+`;
+
+interface TableProps extends React.ComponentPropsWithoutRef<'table'> {
+  data?: Array<any>;
+  columns?:
+    | Array<ReactElement<HeaderRowProps>>
+    | Array<ReactElement<TableHeaderProps> | string>;
+  selectable?: boolean;
+}
+
+export default function Table({
+  columns = [],
+  data = [],
+  selectable: selectableProp = false,
+  children,
+  ...rest
+}: TableProps) {
+  const initialState: State = {
+    sort: {
+      direction: 'asc',
+    },
+    data,
```

Comment:
When the data prop changes how are we applying those new values to the table? I don't see any code that supports that.

Reply:
👍

Reply:
👍 fixed